Transportation Sector Energy Use - nrcan.ipynb

```diff
@@ -0,0 +1,392 @@
+   "metadata": {},
+   "source": [
+    "# Data cleaning"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Imports\n",
+    "Import libraries and write settings here."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 87,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2020-12-07T03:03:36.289681Z",
+     "start_time": "2020-12-07T03:03:35.568612Z"
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The autoreload extension is already loaded. To reload it, use:\n",
+      "  %reload_ext autoreload\n",
+      "The dotenv extension is already loaded. To reload it, use:\n",
+      "  %reload_ext dotenv\n"
+     ]
+    }
+   ],
+   "source": [
+    "%load_ext autoreload\n",
+    "%autoreload\n",
+    "%load_ext dotenv\n",
+    "%dotenv\n",
+    "\n",
+    "import pandas as pd\n",
+    "import os\n",
+    "import qgrid\n",
+    "import ipywidgets\n",
+    "import numpy as np"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Set up extractions from each nrcan csv"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 187,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#extract target name from table name\n",
+    "def substring_table(tableName):\n",
+    "    start = tableName.find(\": \") + len(\": \")\n",
+    "    end = tableName.find(\" Secondary\")\n",
+    "    substring = tableName[start:end]\n",
+    "    return substring\n",
+    "\n",
+    "#Concatenate table_id number\n",
+    "def write_table_id(index):\n",
+    "    table_id = 'tranca2017e_' + str(index)\n",
+    "    return table_id\n",
+    "\n",
+    "#add additional columns to table\n",
+    "def add_target_metadata(df, targetName, table_id):\n",
+    "    df.insert(1, 'Target', targetName)\n",
+    "    df.insert(3, 'Units', 'Petajoules')\n",
+    "    df.insert(4, 'Table ID', table_id)\n",
+    "    return df\n",
+    "\n",
+    "def add_source_metadata(df, sourceName, table_id):\n",
+    "    df.insert(0, 'Source', sourceName)\n",
+    "    df.insert(3, 'Units', 'Petajoules')\n",
+    "    df.insert(4, 'Table ID', table_id)\n",
+    "    return df\n",
+    "\n",
+    "#extract only relevant data\n",
+    "#note locations are hardcodes for nrcan files types -- all are equivalent\n",
+    "def extract_relevant_data(df): \n",
+    "    empty_index = df[df['Name'].isnull()].index.tolist()\n",
+    "    last_i = empty_index[next(x[0] for x in enumerate(empty_index) if x[1] > 10)]\n",
+    "    df = df.drop(df.iloc[:, 2:29], inplace = False, axis = 1)\n",
+    "    df = df.iloc[12:last_i,:]\n",
+    "    df = df.drop(0, axis = 1)\n",
+    "    df , hybrid = find_subtitle_row(df)\n",
+    "    #display(df)\n",
+    "    return df, hybrid\n",
+    "\n",
+    "def find_subtitle_row(df):\n",
+    "    index = df[df['Name'].str.contains(\"(PJ)\")].index\n",
+    "    if index != 12:\n",
```

Comment:
Haven't actually executed this, but do you mean `index[0] != 12`, since (I think) index is a list?